query-engine/connector-test-kit-rs/query-engine-tests/tests/new/regressions/prisma_5952.rs

```diff
@@ -51,4 +51,40 @@ mod regression {
 
         Ok(())
     }
+
+    #[connector_test]
+    async fn decimal_find_different_uniques_unquoted(runner: Runner) -> TestResult<()> {
+        runner
+            .query(indoc! {
+                r#"mutation {createOneArtist(data:{
+                    firstName: "Michael"
+                    netWorth: 236600000.12409
+                }){ firstName }}"#
+            })
+            .await?
+            .assert_success();
+
+        runner
+            .query(indoc! {
+                r#"mutation {createOneArtist(data:{
+                    firstName: "George"
+                    netWorth: -0.23660010012409
+                }){ firstName }}"#
+            })
+            .await?
+            .assert_success();
+
+        let queries = vec![
+            r#"query {findUniqueArtist(where:{firstName_netWorth:{firstName:"Michael",netWorth:236600000.12409}}) {firstName netWorth}}"#.to_string(),
+            r#"query {findUniqueArtist(where:{firstName_netWorth:{firstName:"George",netWorth:-0.23660010012409}}) {firstName netWorth}}"#.to_string(),
+        ];
+
+        let batch_results = runner.batch(queries, false, None).await?;
+        insta::assert_snapshot!(
+            batch_results.to_string(),
+            @r###"{"batchResult":[{"data":{"findUniqueArtist":{"firstName":"Michael","netWorth":"236600000.12409"}}},{"data":{"findUniqueArtist":{"firstName":"George","netWorth":"-0.23660010012409"}}}]}"###
+        );
```

Comment:
Would you mind extracting this function https://github.com/prisma/prisma-engines/blob/e5078ed87490195264685f8f0c7e2a57d0ec3af9/query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/batching/select_one_compound.rs#L387-L409

into `query-engine-tests/src/utils/batch.rs` and re-use it here? I'm mostly interested in asserting that the query document _is_ compacted